codec/reflectcodec/type_codec.go

```diff
@@ -120,82 +115,69 @@ func (c *genericCodec) sizeWithOmitEmpty(value reflect.Value, nullable bool) (in
 	case reflect.String:
 		return wrappers.StringLen(value.String()), false, nil
 	case reflect.Ptr:
-		prefixSize := 0
-		if nullable {
-			if value.IsNil() {
-				return wrappers.BoolLen, false, nil
+		if value.IsNil() {
+			if !nullable {
+				return 0, false, errMarshalNil
 			}
-			prefixSize += wrappers.BoolLen
-		} else if value.IsNil() {
-			return 0, false, errMarshalNil
+			return wrappers.BoolLen, false, nil
 		}
 
-		size, _, err := c.size(value.Elem())
-		if err != nil {
-			return 0, false, err
+		size, constSize, err := c.size(value.Elem(), false)
+		if nullable {
+			return wrappers.BoolLen + size, false, err
```

Comment:
```suggestion
			return wrappers.BoolLen + size, constSize, err
```
?

Reply:
A nullable pointer is not fixed size (because the value could be nil or non-nil)